packages/react/src/components/Storage/FileUploader/UploadTracker/index.tsx

```diff
@@ -64,12 +64,18 @@ export function UploadTracker({
             {name}
           </Text>
         </View>
+        {showEditButton ? (
+          <Button onClick={onStartEdit} size="small" variation="link">
+            <VisuallyHidden>Edit file name {file.name}</VisuallyHidden>
+            <IconEdit aria-hidden fontSize="medium" />
+          </Button>
+        ) : null}
         <Text as="span" className={ComponentClassNames.FileUploaderFileSize}>
           {humanFileSize(size, true)}
         </Text>
       </>
     ),
-    [name, size]
+    [file.name, name, onStartEdit, showEditButton, size]
   );
 
   const Actions = useCallback(() => {
```

Comment:
Just noticed that we are wrapping child components in `useCallback`, usually this hook is useful for handlers/functions, not components, and should be used judiciously as it could actually cause performance issues.
This comment is pertinent to the refactor work, not this PR.

Reply:
Thanks @ioanabrooks  ! Will take a note of it